core/templates/pages/skill-editor-page/modal-templates/add-misconception-modal.directive.html

```diff
@@ -38,6 +38,9 @@ <h3>Add Misconception</h3>
         </div>
         <schema-based-editor class="protractor-test-notes-textarea" schema="MISCONCEPTION_PROPERTY_FORM_SCHEMA" local-value="misconceptionNotes">
         </schema-based-editor>
+        <div ng-if="isMisconceptionNotesHtmlLengthExceeded()" class="oppia-length-validation-error mt-2">
+           Please shorten the content (images, tabs, and other special items take up the most storage).
+        </div>
```

Comment:
Hm, do we actually need to count the characters taken up by actual HTML tags? It might be hard for a user to figure out if their content is under the limit since they may not see the actual HTML tags. @EricZLou could you please check if this is expected?

Reply:
@kevintab95 i discussed the same thing with @EricZLou and he agrees with having html tags included in the count . You can read the top threads to get more insight into this matter .

Reply:
@kevintab95 -- good question; the purpose of having these "HTML length" validation checks it to limit the amount of data that is ultimately stored in our backend storage.

**Previously:** the user could attach 1,000 images and write 2 characters and the validation check would just see a length of 2, even though it is hundreds of thousands of characters that are stored.
**Now:** we limit the HTML length to account for this

Do you think this approach makes sense?

Reply:
Ah, I see. I missed the previous comment / discussion. Sounds good, thanks!

Reply:
Hmmm, echoing my comment, and I don't think I'm convinced yet... Also given this exact message, it is not trivial for the user to find how to get down to the limit. 

I don't think checking the character limit on HTML is really a good idea. The HTML is our internal representation, which can definitely change. We can change the RTE, add new fields in the HTML, or anything like that which would affect the user because of this check. Firstly could you explain why we need this check? And why it makes sense to validate the user input as HTML?

For your concern above, we can have better validations - like limit number of images in the html (or videos). But limiting on the size of the HTML doesn't sound right.

Reply:
I agree with @nithusha21 here. Consider this scenario, the misconception is valid when a user creates it but in the future we do a migration that modifies an existing tag to have more characters than before and that results in the misconception becoming invalid (exceeds the char limit). This seems problematic -- the user won't know what caused it to become invalid.

Reply:
@kevintab95 @nithusha21 @EricZLou I think if we are going to make changes , we need more insight on the approach through which we are going to validate the checks i.e. what to validate checks against ( raw html data, actual content inside html tags or something else ) . Waiting for your response !!

Reply:
Yes @ashish-patwal I definitely agree. As a reviewer I am not comfortable with the changes in this PR as of now. The impact of changing validations is usually drastic if not done correctly so we need to understand a few things. 3 things primarily, what are is the purpose of adding the validation, how can the user easily ensure the input is valid, and would the validation only affect cases when people enter a valid input and then the input becomes invalid after a change on the developer side which is hidden from the user. 

So this being a hard problem by itself, let me ask a few questions that we can try to address and then we can decide on what the right thing to do is...
-  I don't understand the purpose of this validation (as we don't seem to have similar validations in the backend right?). Are we trying to limit the user input to reduce the amount of data we store in the database? Or is there any particular reason apart from this where we want to reduce the size of inputs?
- The validation we add should strictly be on the basis of user input. A user input is either valid or invalid based on what is input. It shouldn't depend on internal representation of the data. As an example, if we would like to limit large images, validate the size of the image added (I think this validation exists). If we would like to limit number of images, limit that specifically. 
- The scheme that we come up with needs to be validated against existing data. If users have saved "invalid" data before your change, it would cause confusion as it would suddenly become invalid. We need to audit and fix these cases.
-  To reiterate, this is not an easy problem, and would not be easy to get right in 1 go. It is one of those cases where it is easy to say this approach is incorrect, but it is much harder to come up with the correct checks. 

Here is my suggestion. As this change is becoming slightly more involved lets start by making a doc (1-2 pages would suffice 😄) which addresses these points, and suggests an approach which we think would work, and you can send it to Kevin, Eric and me. We can discuss over the approach and finalize the strategy that works. Does this sound good?